### PR TITLE
fix: add/remove pausable modifiers from various contracts

### PR DESCRIPTION
**Pull request by aayush-bedrock**

### CODE REVIEW

File: contracts/token.sol [hunk 1]
```diff
@@ -29,7 +29,7 @@ contract BedrockDAO is ERC20, ERC20Burnable, AccessControl, ERC20Pausable, ERC20
         _unpause();
     }
 
-    function mint(address to, uint256 amount) public onlyRole(MINTER_ROLE) {
+    function mint(address to, uint256 amount) public whenNotPaused onlyRole(MINTER_ROLE) {
```
Comment (Trisome-Bedrock) [inline contracts/token.sol:32]: No need to explicitly add 'whenNotPaused' to mint(), as it is already guarded in _beforeTokenTransfer().

File: contracts/penpie_adapter.sol [hunk 2]
```diff
@@ -77,7 +77,7 @@ contract PenpieAdapter is Initializable, OwnableUpgradeable, PausableUpgradeable
     /**
      * @dev updateReward
      */
-    function updateReward() external {  _updateReward(); }
+    function updateReward() external whenNotPaused {  _updateReward(); }
```
Comment (Trisome-Bedrock) [inline contracts/penpie_adapter.sol:80]: Please remove the 'whenNotPaused' modifier, as we now allow the normal distribution of rewards during a pause.

File: contracts/gauge_controller.sol [hunk 3]
```diff
@@ -264,6 +278,7 @@ contract GaugeController is AccessControlUpgradeable, ReentrancyGuardUpgradeable
      */
     function voteForGaugeWeight(address _gAddr, uint256 _userWeight)
         external
+        whenNotPaused
```
Comment (Trisome-Bedrock) [inline contracts/gauge_controller.sol:281]: Please remove the "whenNotPaused" modifier from the voteForGaugeWeight() function. If we make the voteForGaugeWeight function pausable, users might not be able to utilize their already locked voting power from the VotingEscrow contract, resulting in a situation where they can lock but cannot vote.

**Comment by Trisome-Bedrock** [comment]
As you introduced the pausable mechanism, please add '__Pausable_init();' to the [initialize()](https://github.com/Bedrock-Technology/bedrock-dao/blob/575e987a165ba5bccc04e3cc57ca721bce5a8b3c/contracts/gauge_controller.sol#L118) function.

**Comment by Trisome-Bedrock** [comment]
To make the code style more uniform, it is better to have the [GaugeController](https://github.com/Bedrock-Technology/bedrock-dao/blob/575e987a165ba5bccc04e3cc57ca721bce5a8b3c/contracts/gauge_controller.sol#L32) explicitly inherit from the "Initializable" library, requiring the import of "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol".

**Comment by Trisome-Bedrock** [comment]
Please remove the 'whenNotPaused' modifier from the [distributeRewards()](https://github.com/Bedrock-Technology/bedrock-dao/blob/575e987a165ba5bccc04e3cc57ca721bce5a8b3c/contracts/cashier.sol#L95C70-L95C83) function, as we have decided not to restrict reward distribution during a pause.

**Comment by Trisome-Bedrock** [comment]
Please ensure that all related [tests](https://github.com/Bedrock-Technology/bedrock-dao/tree/575e987a165ba5bccc04e3cc57ca721bce5a8b3c/tests) pass after code changes.

**Review by Zealy-Rockx — APPROVED** [review]

**Review by Zealy-Rockx — APPROVED** [review]
LGTM

**Review by Trisome-Bedrock — APPROVED** [review]

**Pull request merged**